src/main/java/com/fc/invoicing/services/CompanyService.java

```diff
@@ -0,0 +1,71 @@
+package com.fc.invoicing.services;
+
+import com.fc.invoicing.db.Database;
+import com.fc.invoicing.db.JpaCompanyRepository;
+import com.fc.invoicing.dto.CompanyDto;
+import com.fc.invoicing.dto.mappers.CompanyListMapper;
+import com.fc.invoicing.dto.mappers.CompanyMapper;
+import com.fc.invoicing.exeptions.handlers.ItemNotFoundExemption;
+import com.fc.invoicing.model.Company;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@AllArgsConstructor
+@Service
+public class CompanyService implements Database<CompanyDto> {
+
+    private final JpaCompanyRepository jpaCompanyRepository;
+    private final CompanyListMapper companyListMapper;
+    private final CompanyMapper companyMapper;
+
+    @Override
+    public CompanyDto add(CompanyDto company) {
+        company.setCompanyId(UUID.randomUUID());
+        return companyMapper.toDto(jpaCompanyRepository.save(companyMapper.toEntity(company)));
+    }
+
+    @Override
+    public CompanyDto getById(UUID id) {
+        if (jpaCompanyRepository.findById(id).isPresent()) {
```

Comment:
try to simplify it with jpaCompanyRepository.findById(id).map().orElseThrow()

Reply:
changed